params.go

```diff
@@ -117,3 +156,55 @@ func (b *paramBuilder) addInt32(name *C.char, value int32) {
 		b.err = newOpenSSLError("OSSL_PARAM_BLD_push_int32(" + C.GoString(name) + ")")
 	}
 }
+
+// addBN adds a GO_BIGNUM_PTR to the builder.
+func (b *paramBuilder) addBN(name *C.char, value C.GO_BIGNUM_PTR) {
+	if !b.check() {
+		return
+	}
+	if C.go_openssl_OSSL_PARAM_BLD_push_BN(b.bld, name, value) != 1 {
+		b.err = newOpenSSLError("OSSL_PARAM_BLD_push_BN(" + C.GoString(name) + ")")
+	}
+}
+
+// addBin adds a byte slice to the builder.
+// The slice is converted to a BIGNUM using BN_bin2bn and freed when the builder is finalized.
+// If private is true, the BIGNUM will be cleared with BN_clear_free,
+// otherwise it will be freed with BN_free.
+func (b *paramBuilder) addBin(name *C.char, value []byte, private bool) {
+	if !b.check() {
+		return
+	}
+	if value == nil {
```

Comment:
When I see this, I have to wonder if distinguishing nil vs. empty slices could bite us later in subtle ways... e.g. if an OpenSSL param behaves differently if passed an empty array vs. not being passed at all. (An unexpected default value?)

I suppose the alternatives are:
Should we actually pass a reference to a zero-length array if `value` is nil, and make the caller responsible for skipping the call if it determines it's possible?
Or: if it's always ok to not pass a nil/0-length param, should this just check `if len(value) == 0`?

Reply:
Good point. Go best practices says not to distinguish between an empty and a nil slice unless necessary. I'll do that here.